godel_scan_analysis/src/keyence_scan_server.cpp

```diff
@@ -0,0 +1,81 @@
+#include "godel_scan_analysis/keyence_scan_server.h"
+
+#include <pcl_ros/transforms.h>
+#include <pcl/filters/voxel_grid.h>
+
+// Constants
+const static double VOXEL_GRID_LEAF_SIZE = 0.005; // 5 mm
+
+const static double VOXEL_GRID_PUBLISH_PERIOD = 2.0; // seconds
+
+const static double TF_WAIT_TIMEOUT = 0.25; // seconds
+
+godel_scan_analysis::ScanServer::ScanServer(const std::string& world_frame, 
+                                            const std::string& scan_frame)
+  : map_(new ColorCloud)
+  , buffer_(new ColorCloud)
+  , from_frame_(world_frame)
+  , to_frame_(scan_frame)
+{
+  ros::NodeHandle nh;
+  scan_sub_ = nh.subscribe("profiles", 500, &ScanServer::scanCallback, this);
+  cloud_pub_ = nh.advertise<ColorCloud>("color_cloud", 1);
```

Comment:
Quick comment: should they? Making topic names constant strings is not a common practice in ROS nodes, as it removes the name of the topic being *advertised* from the context in which the advertising publisher is created (making it harder to read the code). In addition, users are not supposed to change topic names in nodes by changing constants and then recompiling, but by using *remapping*. Finaly, topic names are mostly only defined once (ie: where the publisher is created), and not reused in other code.

Consistency with other nodes in Godel is probably more important, but I feel thinking about this a bit makes sense.